code/network/multiutil.cpp

```diff
@@ -757,16 +751,16 @@ int multi_create_player( int net_player_num, player *pl, const char* name, net_a
 
 		// find the ship that matches the string stored in default_player_ship
 
-		for (i = 0; i < Num_ship_classes; i++) {
-			if ( !stricmp(Ship_info[i].name, default_player_ship) ) {
-				player_ship_class = i;
+		for (auto it = Ship_info.cbegin(); it != Ship_info.cend(); ++it) {
+			if ( !stricmp(it->name, default_player_ship) ) {
+				player_ship_class = std::distance(Ship_info.cbegin(), it);
 				break;
 			}
 		}
 
-		if (i == Num_ship_classes)
+		if (player_ship_class == -1)	// if ship_class is -1 (as the condition to enter this branch at all means), then player_ship_class is initialized to -1 as well.
```

Comment:
The `if ( ship_class == -1)` conditional was there before. The logic is equivalent.